src/features/reviews/utils/adjustQueue.js

```diff
@@ -9,15 +9,35 @@
 
 import _ from 'lodash';
 
-import { MEANING, RADICAL } from 'src/common/constants';
+import { MEANING, RADICAL, RANDOM_ORDER, LOWEST_LEVEL_FIRST, CURRENT_LEVEL_FIRST, ASCENDING_SRS_STAGE, DESCENDING_SRS_STAGE } from 'src/common/constants';
 
 const ALLOWED_MAX_DISTANCE = 10;
 const ALLOWED_MIN_DISTANCE = 3;
 
-export default (queue, backToBackMode = false, meaningFirst = false) => {
+export default (queue, backToBackMode = false, meaningFirst = false, reviewOrder = RANDOM_ORDER) => {
+
   const newQueue = queue.slice();
   let currentIndex = 0;
 
+  // sort new queue based on the selected review order
+  switch (reviewOrder) {
+    case RANDOM_ORDER:
+      newQueue.sort(() => Math.random() - 0.5);
```

Comment:
We already have `lodash` as a dependency, you can use [_.shuffle](https://lodash.com/docs/4.17.15#shuffle) to randomise array order; something like:

```javascript
newQueue = _.shuffle(newQueue)
```

Reply:
That's cool, and is a way better shuffling algorithm too! Thanks for the suggestion :)